drivers/storage/class/class2/class2.c

```diff
@@ -828,19 +828,19 @@ Return Value:
             lastSector + 1));
 
         //
-        // Calculate media capacity in bytes.
+        // Calculate number of cylinders.
         //
 
-        deviceExtension->PartitionLength.QuadPart = (LONGLONG)(lastSector + 1);
+        deviceExtension->DiskGeometry->Geometry.Cylinders.QuadPart = (LONGLONG)((lastSector + 1)/(DEFAULT_SECTORS_PER_TRACK * DEFAULT_TRACKS_PER_CYLINDER));
 
         //
-        // Calculate number of cylinders.
+        // Calculate media capacity in bytes.
         //
 
-        deviceExtension->DiskGeometry->Geometry.Cylinders.QuadPart = (LONGLONG)((lastSector + 1)/(DEFAULT_SECTORS_PER_TRACK * DEFAULT_TRACKS_PER_CYLINDER));
-
+        deviceExtension->PartitionLength.QuadPart = (LONGLONG)(lastSector + 1);
         deviceExtension->PartitionLength.QuadPart =
```

Comment:
Any idea why this is not just `(lastSector + 1) << deviceExtension->SectorShift`?
Anyway, I'm merging this because the days of this driver are counted

Reply:
No good ideas. I just kept it that way to keep the changes minimal.